enterprise/models.py

```diff
@@ -4723,6 +4724,24 @@ class Meta:
         unique_together = (("name", "enterprise_customer"),)
         ordering = ['-modified']
 
+    def delete(self, *args, **kwargs):
+        """
+        A psuedo-post delete signal since this is a SoftDeletableModal that removes any
+        PolicyGroupAssocations from enterprise-access assocated with the deleted group
+        """
+        self.is_removed = True
+        self.save(update_fields=['is_removed'])
```

Comment:
We should be able to leverage the base class's delete functionality here, which should take care of this operation.
```suggestion
        super().delete(*args, **kwargs)
```